storage/innobase/include/data0type.h

```diff
@@ -315,6 +320,16 @@ ulint dtype_get_at_most_n_mbchars(
  @return true if string type */
 bool dtype_is_string_type(
     ulint mtype); /*!< in: InnoDB main data type code: DATA_CHAR, ... */
+
+#ifdef XTRABACKUP
+/** Checks if a data main type is a real temporal type.
+ @return true if real temporal type */
```

Comment:
@doxygen style comments please. What is real and not real? can u explain in comment?

Reply:
From what I have seen and studied, `real type` has different meanings in different contexts. It can mean

- a data type allowing fractional numbers for precision
- an underlying datatype used internally. For example binlog stores  CHAR, VARCHAR, SET and ENUM as MYSQL_TYPE_STRING. For such types, `field->type()` will be MYSQL_TYPE_STRING and `field->real_type()` will return the actual underlying data type.

However in the context of collation, temporal real types are the ones that have fractional numbers for precision as they need to be stored and displayed as FIXBINARY. So, it is safe to assume that `real temporal type` are the types which allow with fractional seconds.